cql3/restrictions/statement_restrictions.hh

```diff
@@ -41,24 +42,30 @@ private:
     /**
      * Restrictions on partitioning columns
      */
-    ::shared_ptr<partition_key_restrictions> _partition_key_restrictions;
+    expr::expression _new_partition_key_restrictions;
```

Comment:
That name is not future-proof: if we ever change the representation again, we'll only be able to use `newer` or `newest`. `_partition_key_restrictions_expression` maybe?

Same for all `_new_X` below